### PR TITLE
use setSelection from SelectionPlugin

### PR DESCRIPTION
**Pull request by FrancoisGe**

### CODE REVIEW

File: addons/html_editor/static/src/editor/core/delete_plugin.js [hunk 1]
```diff
@@ -38,8 +40,6 @@ import {
     rightPos,
 } from "../utils/position";
 import { getDeepRange, preserveCursor, setSelection } from "../utils/selection";
```
Comment (ged-odoo) [inline addons/html_editor/static/src/editor/core/delete_plugin.js:42]: you keep the import?

File: addons/html_editor/static/src/editor/table/table_plugin.js [hunk 2]
```diff
@@ -289,7 +294,11 @@ export class TablePlugin extends Plugin {
         if (!cursorDestination) {
             return false;
         }
-        setSelection(...startPos(cursorDestination), ...endPos(cursorDestination), true);
+        this.shared.setSelection(
+            ...startPos(cursorDestination),
+            ...endPos(cursorDestination),
+            true
```
Comment (ged-odoo) [inline addons/html_editor/static/src/editor/table/table_plugin.js]: remove the `true`, it's the default now

File: addons/html_editor/static/src/editor/core/delete_plugin.js (hunk 1 above)
Comment (FrancoisGe) [reply addons/html_editor/static/src/editor/core/delete_plugin.js:42]: I need it for an utils in the file. We should maybe move the utils in the plugin

File: addons/html_editor/static/src/editor/table/table_plugin.js (hunk 2 above)
Comment (FrancoisGe) [reply addons/html_editor/static/src/editor/table/table_plugin.js]: Yes

**Pull request merged**